### PR TITLE
Support for PNG monochromatic images

### PR DESCRIPTION
**Pull request by GerardoAllende**
Just a little patch to support monochromatic images. If img.mode == "1" then it's supported now. Saves some PDF bytes with QR codes.

**Checklist**:

- [x] The GitHub pipeline is OK (green), 
      meaning that both `pylint` (static code analyzer) and `black` (code formatter) are happy with the changes of this PR.

- [ ] A unit test is covering the code added / modified by this PR

- [x] This PR is ready to be merged 

- [ ] In case of a new feature, docstrings have been added, with also some documentation in the `docs/` folder

- [ ] A mention of the change is present in `CHANGELOG.md`

By submitting this pull request, I confirm that my contribution is made under the terms of the [GNU LGPL 3.0 license](https://github.com/PyFPDF/fpdf2/blob/master/LICENSE).

### CODE REVIEW

File: fpdf/image_parsing.py [hunk 1]
```diff
@@ -171,8 +174,12 @@ def _to_zdata(img, remove_slice=None, select_slice=None):
     if select_slice:
         data = data[select_slice]
     # Left-padding every row with a single zero:
-    channels_count = len(data) // (img.size[0] * img.size[1])
-    loop_incr = img.size[0] * channels_count + 1
+    if img.mode == '1':
+        import math
```
Comment (Lucas-C) [inline fpdf/image_parsing.py:178]: Please move this `import` at the top of this file

**Review by Lucas-C — CHANGES_REQUESTED** [review]
Could you please add a mention of this in `CHANGELOG.md` and a unit test?

I suggest the following approach: taking inpiration from the existing test functions,
write a basic usage scenario that calls `assert_pdf_equal(pdf, "new_test_name.pdf", tmp_path, generate=True)`, and run `pytest -vv test/shape/test_arc.py -k new_test_name`
The `generate=True` argument is very useful to write new tests as it will generate the reference PDF file.
Once you'll be happy with the PDF file generated, you will just have to remove `generate=True` and you'll get a fine non-regression unit test.

File: fpdf/image_parsing.py (hunk 1 above)
Comment (GerardoAllende) [reply fpdf/image_parsing.py]: Done

**Comment by Lucas-C** [comment]
I see you have addressed my review comment @GerardoAllende, thank you!

Have you seen my other request regarding this PR?

> Could you please add a mention of this in `CHANGELOG.md` and a unit test?

**Pull request closed without merging**